Base/src/main/java/io/deephaven/base/ringbuffer/CharRingBuffer.java

```diff
@@ -144,15 +155,14 @@ public void ensureRemaining(int count) {
     }
 
     /**
-     * Add values unsafely (will silently overwrite values if the buffer is full). This call should be used in
-     * conjunction with {@link #ensureRemaining(int)}.
+     * Add values without overflow detection. Making this call when the buffer is full will result in this data
+     * structure becoming corrupted and unusable. This call must be used in conjunction with
+     * {@link #ensureRemaining(int)} or {@link #remaining()} to verify remaining capacity if sufficient.
      *
      * @param e the value to add to the buffer
      */
     public void addUnsafe(char e) {
-        storage[tail] = e;
-        tail = (tail + 1) % storage.length;
-        size++;
+        storage[(int) (tail++ & mask)] = e;
```

Comment:
I forgot to ask. Have we decided on a policy (ignore vs explicitly check) for int64 wraparound?

Reply:
Ignore, considering that we will reset with every call to grow().

Reply:
IMO, that doesn't sound like a valid rationale to me. A ring buffer only grows a few times in its lifetime, but on the other hand can be in a steady state (adding/removing data) for an arbitrarily long time without ever growing.

I'm actually ok with not checking, but with the right rationale: namely that that int64s are really big numbers and they're not going to wrap during any particular run of the program.

I'm also ok with checking.

Reply:
I added some code to the safe `add()` operations and to `remaining()` and `ensureRemaining()`.  I'm probably trying to hard to keep the unsafe code streamlined, I can be convinced to put this in `addUnsafe()` and trust the predictor.  Lemme know what you think.